analytic_engine/src/sst/parquet/meta_data.rs

```diff
@@ -117,23 +117,35 @@ impl Filter for Xor8Filter {
 
 pub struct RowGroupFilterBuilder {
     builders: Vec<Option<Xor8Builder>>,
+    column_values: Vec<HashSet<Vec<u8>>>,
+    min_value_num_when_build_filter: usize,
 }
 
 impl RowGroupFilterBuilder {
-    pub(crate) fn with_num_columns(num_col: usize) -> Self {
+    pub(crate) fn with_num_columns(num_col: usize, min_value_num_when_build_filter: usize) -> Self {
         Self {
             builders: vec![None; num_col],
+            column_values: vec![HashSet::new(); num_col],
+            min_value_num_when_build_filter,
         }
     }
 
     pub(crate) fn add_key(&mut self, col_idx: usize, key: &[u8]) {
-        self.builders[col_idx].get_or_insert_default().insert(key)
+        self.builders[col_idx].get_or_insert_default().insert(key);
+        // TODO: remove to_owned?
+        self.column_values[col_idx].insert(key.to_owned());
```

Comment:
I guess Xor8Builder can provide an estimate for the number of unique keys, and we can use that instead of an extra `HashSet`.

And I guess a better way is to pre-compute this number of unique keys, and to avoid building xorfilter if it should not. However, maybe we can add metrics to figure out the proportion of the total which doesn't need bloom filter.